src/librustdoc/html/static/js/search.js

```diff
@@ -3540,6 +3552,28 @@ class DocSearch {
             return false;
         }
 
+        /**
+         * This function checks if the given list contains any
+         * types mentioned in the query.
+         *
+         * @param {Array<FunctionType>} list    - A list of function types.
+         * @param {[FunctionType]} where_clause - Trait bounds for generic items.
+         */
+        function containsTypeFromQuery(list, where_clause) {
```

Comment:
Wouldn't this catch builder patterns like `fn set_some_var(self) -> Self`? Maybe that is fine but it seems like it might be better to just match `Clone::clone` if possible. Possibly `Default::default` for a similar reason?

Reply:
That's very intentional.  As I see it, the semantic meaning of `-> SomeType` is "how do I get `SomeType`".  While a pure functional builder pattern does not technically mutate, it is not a satisfactory answer to "how do i get the thing".

Also remember that:
1. We're not filtering them out, only ranking them lower.  For simple types that have few methods, they're still easily accessible.
2. result sorting is a zero-sum game, something has to go at the bottom, and something has to go at the top.

However, `Default::default` is intentionally *not* deprioritized, as it *is* a reasonable answer to "how do i get the thing", and in some crates, it may be the only constructor.

Hope this clears things up :]

Reply:
I suppose I didn't do a good job of explaining what I meant by "clone-like": i specifically mean functions that take a value of some type, and then return another value of that type, similar to an identity function, except that it may take more arguments.